src/middlewared/middlewared/etc_files/nfsd.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+
+from middlewared.utils import run
+
+logger = logging.getLogger(__name__)
+
+
+async def get_exports(config, shares, kerberos_keytabs):
+    networks_pool = defaultdict(lambda: defaultdict(lambda: 0))
+
+    result = []
+
+    if config["v4"]:
+        if config["v4_krb"] or kerberos_keytabs:
+            result.append("V4: / -sec=krb5:krb5i:krb5p")
+        else:
+            result.append("V4: / -sec=sys")
+
+    for share in shares:
+        if share["paths"]:
+            share = build_share(config, share, networks_pool)
+            if share:
+                result.append(share)
+
+    return "\n".join(result) + "\n"
+
+
+def build_share(config, share, networks_pool):
+    if share["paths"]:
+        result = list(share["paths"])
+
+        if share["alldirs"]:
+            result.append("-alldirs")
+
+        if share["ro"]:
+            result.append("-ro")
+
+        if share["quiet"]:
+            result.append("-quiet")
+
+        if share["mapall_user"] and share["mapall_group"]:
+            result.append(f'-mapall="{share["mapall_user"]}":"{share["mapall_group"]}"')
+        elif share["maproot_user"] and share["maproot_group"]:
+            result.append(f'-maproot="{share["maproot_user"]}":"{share["maproot_group"]}"')
+
+        if config["v4"] and share["security"]:
+            result.append("-sec=" + ":".join([s.lower() for s in share["security"]]))
+
+        targets = build_share_targets(share, networks_pool)
+
+        if targets:
+            result.extend(targets)
+
+            return " ".join(result)
+
+
+def build_share_targets(share, networks_pool):
+    result = []
+
+    try:
+        dev = os.stat(share["paths"][0]).st_dev
+    except Exception as e:
+        logger.warning("Unable to stat {share['paths'][0]:r}: {e}")
+    else:
+        for network in share["networks"]:
+            try:
+                network = ipaddress.ip_network(network, strict=False)
+            except Exception as e:
+                logger.warning(f"Invalid network: {network} ({e})")
+            else:
+                inc = networks_pool[dev][network]
+                networks_pool[dev][network] += 1
+                if networks_pool[dev][network] > 2 ** (32 - network.prefixlen):
+                    logger.warning(f"No space for network {network} on path {share['paths'][0]}")
+                    continue
+
+                result.append("-network " + str(network.network_address + inc) + "/" + str(network.prefixlen))
+
+        for host in share["hosts"]:
+            try:
+                network = ipaddress.ip_network(f"{host}/32")
+            except Exception as e:
+                logger.warning(f"Invalid IP address: {host} ({e})")
+            else:
+                networks_pool[dev][network] += 1
+                if networks_pool[dev][network] > 2 ** (32 - network.prefixlen):
+                    logger.warning(f"No space for host {host} on path {share['paths'][0]}")
+                    continue
```

Comment:
Does this make sense for hosts? If its /32 there should be only one possible, no?

Reply:
This is copypaste so we handle `192.168.0.1` and `192.168.0.1/32` as same thing (as nfsd does)

Reply:
Same thing but as duplicated code, so why bother doing the extra math? Isnt /32 always allowed anyway unless duplicated?